cc-wrapper: Fix for prebuilt android

Ericson2314 opened the pull request:
###### Motivation for this change

We don't want to use Nix-built GCC's libs with prebuilt clang in this case.

CC @bkchr & @matthewbauer 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Ericson2314 committed:
cc-wrapper: Fix for prebuilt android

We don't want to use Nix-built GCC's libs with prebuilt clang in this
case.

bkchr commented:
@Ericson2314 did you tried this?

I still get the following error:
```
unpacking sources
patching sources
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-
patching script interpreter paths in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-
checking for references to /build/ in /nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-...
substituteStream(): WARNING: pattern '
' doesn't match anything in file '/nix/store/q3wdzrk9vran5mzzd9pcqyl43i1d90zw-aarch64-unknown-linux-android-ndk-gcc-binutils-wrapper-/nix-support/libc-ldflags'
building '/nix/store/i3hxsq3awwvam4pl95hc3iw4cn8dppr0-stdenv-linux.drv'...
building '/nix/store/ai01wnlh8g40phl3h378dil3x3z1y84a-hello-2.10-aarch64-unknown-linux-android.drv'...
unpacking sources
unpacking source archive /nix/store/3x7dwzq014bblazs7kq20p9hyzz0qh8g-hello-2.10.tar.gz
source root is hello-2.10
setting SOURCE_DATE_EPOCH to timestamp 1416139241 of file hello-2.10/ChangeLog
patching sources
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
configuring
configure flags: --disable-dependency-tracking --prefix=/nix/store/s15p212cw9kjyy6m8klshd1rzcilnbc8-hello-2.10-aarch64-unknown-linux-android --build=x86_64-unknown-linux-gnu --host=aarch64-unknown-linux-android
checking for a BSD-compatible install... /nix/store/2f6zzvixj3wsj81flbdcpc6fdscgx4ai-coreutils-8.31/bin/install -c
checking whether build environment is sane... yes
checking for aarch64-unknown-linux-android-strip... aarch64-unknown-linux-android-strip
checking for a thread-safe mkdir -p... /nix/store/2f6zzvixj3wsj81flbdcpc6fdscgx4ai-coreutils-8.31/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for aarch64-unknown-linux-android-gcc... aarch64-unknown-linux-android-clang
checking whether the C compiler works... no
configure: error: in `/build/hello-2.10':
configure: error: C compiler cannot create executables
See `config.log' for more details
builder for '/nix/store/ai01wnlh8g40phl3h378dil3x3z1y84a-hello-2.10-aarch64-unknown-linux-android.drv' failed with exit code 77
cannot build derivation '/nix/store/dfy133728wcyf0ghmkiz356jjzv2ff8q-shell.drv': 1 dependencies couldn't be built
error: build of '/nix/store/dfy133728wcyf0ghmkiz356jjzv2ff8q-shell.drv' failed
```

Or is this something different?

Ericson2314 committed:
deployAndroidPackage: Use `pname` and `version`

Ericson2314 committed:
androidndkPkgs: Take version from underlying package

We can now do this, so let's.

Ericson2314 committed:
lib.systems.examples: Bump android SDK to 21

074bc78cc8749faa31729096b65f2ef51b10abeb evidently meant to do this, but
forgot.

Ericson2314 committed:
androidndk: Avoid depending on target-sensative openjdk

We just need jdk for this script that helps set up toolchains. The
script might need to know about the target platform, but the interpreter
that runs it (or part of it) doesn't.

Ericson2314 commented:
Oops!! I fixed the thing I found, but forgot to do the end-to-end test :flushed:. Did that now, fixed the problem (version number in `lib/systems/examples.nix`), and also cleaned up a few other things.
